gov2/iam/scenarios/scenario_assume_role.go

```diff
@@ -0,0 +1,311 @@
+	return AssumeRoleScenario{
+		sdkConfig: 		sdkConfig,
+		accountWrapper: actions.AccountWrapper{IamClient: iamClient},
+		policyWrapper:  actions.PolicyWrapper{IamClient: iamClient},
+		roleWrapper:    actions.RoleWrapper{IamClient: iamClient},
+		userWrapper:    actions.UserWrapper{IamClient: iamClient},
+		questioner:     questioner,
+		helper:         helper,
+	}
+}
+
+// addTestOptions appends the API options specified in the original configuration to
+// another configuration. This is used to attach the middleware stubber to clients
+// that are constructed during the scenario, which is needed for unit testing.
+func (scenario AssumeRoleScenario) addTestOptions(scenarioConfig *aws.Config) {
+	if scenario.isTestRun {
+		scenarioConfig.APIOptions = append(scenarioConfig.APIOptions, scenario.sdkConfig.APIOptions...)
+	}
+}
+
+// Run runs the interactive scenario.
+func (scenario AssumeRoleScenario) Run() {
+	defer func() {
+		if r := recover(); r != nil {
+			log.Printf("Something went wrong with the demo.\n")
+			log.Println(r)
+		}
+	}()
+
+	log.Println(strings.Repeat("-", 88))
+	log.Println("Welcome to the AWS Identity and Access Management (IAM) assume role demo.")
+	log.Println(strings.Repeat("-", 88))
+
+	user := scenario.CreateUser()
+	accessKey := scenario.CreateAccessKey(user)
+	role := scenario.CreateRoleAndPolicies(user)
+	noPermsConfig := scenario.ListBucketsWithoutPermissions(accessKey)
+	scenario.ListBucketsWithAssumedRole(noPermsConfig, role)
+	scenario.Cleanup(user, role)
+
+	log.Println(strings.Repeat("-", 88))
+	log.Println("Thanks for watching!")
+	log.Println(strings.Repeat("-", 88))
+}
+
+// CreateUser creates a new IAM user. This user has no permissions.
+func (scenario AssumeRoleScenario) CreateUser() *types.User {
+	log.Println("Let's create an example user with no permissions.")
+	userName := scenario.questioner.Ask("Enter a name for the example user:", demotools.NotEmpty{})
+	user, err := scenario.userWrapper.GetUser(userName)
+	if err != nil {
+		panic(err)
+	}
+	if user == nil {
+		user, err = scenario.userWrapper.CreateUser(userName)
+		if err != nil {
+			panic(err)
+		}
+		log.Printf("Created user %v.\n", *user.UserName)
+	} else {
+		log.Printf("User %v already exists.\n", *user.UserName)
+	}
+	log.Println(strings.Repeat("-", 88))
+	return user
+}
+
+// CreateAccessKey creates an access key for the user.
+func (scenario AssumeRoleScenario) CreateAccessKey(user *types.User) *types.AccessKey {
+	accessKey, err := scenario.userWrapper.CreateAccessKeyPair(*user.UserName)
+	if err != nil {
+		panic(err)
+	}
+	log.Printf("Created access key %v for your user.", *accessKey.AccessKeyId)
+	log.Println("Waiting a few seconds for your user to be ready...")
+	scenario.helper.Pause(10)
+	log.Println(strings.Repeat("-", 88))
+	return accessKey
+}
+
+// CreateRoleAndPolicies creates a policy that grants permission to list S3 buckets for
+// the current account and attached it to a newly created role. It also adds an inline
```

Comment:
Should "attached it" be "attaches the policy"?

Reply:
fixed